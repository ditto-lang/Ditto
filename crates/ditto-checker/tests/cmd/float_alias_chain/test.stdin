module Test exports (..);

type alias A = Float;
type alias B = A;
type alias C = B;

five : C = 5.0;
