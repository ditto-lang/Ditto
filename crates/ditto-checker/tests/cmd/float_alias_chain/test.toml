bin.name = "ditto-checker-testbin-check-module"
args = ["float_alias_chain.ditto"]
fs.sandbox = true
