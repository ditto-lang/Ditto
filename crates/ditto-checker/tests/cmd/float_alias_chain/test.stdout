{
  "module_name": [
    "Test"
  ],
  "exports": {
    "types": {
      "A": {
        "Alias": {
          "doc_comments": [],
          "doc_position": 0,
          "kind": "Type",
          "aliased_type": {
            "type": "PrimConstructor",
            "data": "Float"
          },
          "alias_variables": []
        }
      },
      "B": {
        "Alias": {
          "doc_comments": [],
          "doc_position": 1,
          "kind": "Type",
          "aliased_type": {
            "type": "ConstructorAlias",
            "data": {
              "constructor_kind": "Type",
              "canonical_value": {
                "module_name": [
                  null,
                  [
                    "Test"
                  ]
                ],
                "value": "A"
              },
              "source_value": {
                "module_name": null,
                "value": "A"
              },
              "alias_variables": [],
              "aliased_type": {
                "type": "PrimConstructor",
                "data": "Float"
              }
            }
          },
          "alias_variables": []
        }
      },
      "C": {
        "Alias": {
          "doc_comments": [],
          "doc_position": 2,
          "kind": "Type",
          "aliased_type": {
            "type": "ConstructorAlias",
            "data": {
              "constructor_kind": "Type",
              "canonical_value": {
                "module_name": [
                  null,
                  [
                    "Test"
                  ]
                ],
                "value": "B"
              },
              "source_value": {
                "module_name": null,
                "value": "B"
              },
              "alias_variables": [],
              "aliased_type": {
                "type": "ConstructorAlias",
                "data": {
                  "constructor_kind": "Type",
                  "canonical_value": {
                    "module_name": [
                      null,
                      [
                        "Test"
                      ]
                    ],
                    "value": "A"
                  },
                  "source_value": {
                    "module_name": null,
                    "value": "A"
                  },
                  "alias_variables": [],
                  "aliased_type": {
                    "type": "PrimConstructor",
                    "data": "Float"
                  }
                }
              }
            }
          },
          "alias_variables": []
        }
      }
    },
    "constructors": {},
    "values": {
      "five": {
        "doc_comments": [],
        "doc_position": 0,
        "value_type": {
          "type": "ConstructorAlias",
          "data": {
            "constructor_kind": "Type",
            "canonical_value": {
              "module_name": [
                null,
                [
                  "Test"
                ]
              ],
              "value": "C"
            },
            "source_value": {
              "module_name": null,
              "value": "C"
            },
            "alias_variables": [],
            "aliased_type": {
              "type": "ConstructorAlias",
              "data": {
                "constructor_kind": "Type",
                "canonical_value": {
                  "module_name": [
                    null,
                    [
                      "Test"
                    ]
                  ],
                  "value": "B"
                },
                "source_value": {
                  "module_name": null,
                  "value": "B"
                },
                "alias_variables": [],
                "aliased_type": {
                  "type": "ConstructorAlias",
                  "data": {
                    "constructor_kind": "Type",
                    "canonical_value": {
                      "module_name": [
                        null,
                        [
                          "Test"
                        ]
                      ],
                      "value": "A"
                    },
                    "source_value": {
                      "module_name": null,
                      "value": "A"
                    },
                    "alias_variables": [],
                    "aliased_type": {
                      "type": "PrimConstructor",
                      "data": "Float"
                    }
                  }
                }
              }
            }
          }
        }
      }
    }
  },
  "types": {
    "A": {
      "Alias": {
        "doc_comments": [],
        "type_name_span": {
          "start_offset": 37,
          "end_offset": 38
        },
        "kind": "Type",
        "aliased_type": {
          "type": "PrimConstructor",
          "data": "Float"
        },
        "alias_variables": []
      }
    },
    "B": {
      "Alias": {
        "doc_comments": [],
        "type_name_span": {
          "start_offset": 58,
          "end_offset": 59
        },
        "kind": "Type",
        "aliased_type": {
          "type": "ConstructorAlias",
          "data": {
            "constructor_kind": "Type",
            "canonical_value": {
              "module_name": [
                null,
                [
                  "Test"
                ]
              ],
              "value": "A"
            },
            "source_value": {
              "module_name": null,
              "value": "A"
            },
            "alias_variables": [],
            "aliased_type": {
              "type": "PrimConstructor",
              "data": "Float"
            }
          }
        },
        "alias_variables": []
      }
    },
    "C": {
      "Alias": {
        "doc_comments": [],
        "type_name_span": {
          "start_offset": 75,
          "end_offset": 76
        },
        "kind": "Type",
        "aliased_type": {
          "type": "ConstructorAlias",
          "data": {
            "constructor_kind": "Type",
            "canonical_value": {
              "module_name": [
                null,
                [
                  "Test"
                ]
              ],
              "value": "B"
            },
            "source_value": {
              "module_name": null,
              "value": "B"
            },
            "alias_variables": [],
            "aliased_type": {
              "type": "ConstructorAlias",
              "data": {
                "constructor_kind": "Type",
                "canonical_value": {
                  "module_name": [
                    null,
                    [
                      "Test"
                    ]
                  ],
                  "value": "A"
                },
                "source_value": {
                  "module_name": null,
                  "value": "A"
                },
                "alias_variables": [],
                "aliased_type": {
                  "type": "PrimConstructor",
                  "data": "Float"
                }
              }
            }
          }
        },
        "alias_variables": []
      }
    }
  },
  "constructors": {},
  "values": {
    "five": {
      "doc_comments": [],
      "name_span": {
        "start_offset": 82,
        "end_offset": 86
      },
      "expression": {
        "expression": "Float",
        "data": {
          "span": {
            "start_offset": 93,
            "end_offset": 96
          },
          "value": "5.0",
          "value_type": {
            "type": "ConstructorAlias",
            "data": {
              "constructor_kind": "Type",
              "canonical_value": {
                "module_name": [
                  null,
                  [
                    "Test"
                  ]
                ],
                "value": "C"
              },
              "source_value": {
                "module_name": null,
                "value": "C"
              },
              "alias_variables": [],
              "aliased_type": {
                "type": "ConstructorAlias",
                "data": {
                  "constructor_kind": "Type",
                  "canonical_value": {
                    "module_name": [
                      null,
                      [
                        "Test"
                      ]
                    ],
                    "value": "B"
                  },
                  "source_value": {
                    "module_name": null,
                    "value": "B"
                  },
                  "alias_variables": [],
                  "aliased_type": {
                    "type": "ConstructorAlias",
                    "data": {
                      "constructor_kind": "Type",
                      "canonical_value": {
                        "module_name": [
                          null,
                          [
                            "Test"
                          ]
                        ],
                        "value": "A"
                      },
                      "source_value": {
                        "module_name": null,
                        "value": "A"
                      },
                      "alias_variables": [],
                      "aliased_type": {
                        "type": "PrimConstructor",
                        "data": "Float"
                      }
                    }
                  }
                }
              }
            }
          }
        }
      }
    }
  },
  "values_toposort": [
    "five"
  ]
}